### PR TITLE
37 frontend display total units

### PR DESCRIPTION
**Pull request by jluna28**

### CODE REVIEW

File: src/frontend/src/Components/Grid.tsx [hunk 1]
```diff
@@ -92,12 +105,17 @@ function Grid() {
...
           flowchart ? (
-            flowchart.map((term: TermData) => {
+            flowchart.map((term: TermData, index: number) => {
               const classes: ClassDBClass[] =
                 term.classes.map((flowchartClass: FlowchartClass) => classDB[flowchartClass.id]);
               return (
                 <div className='term' key={term.termName}>
-                  <Term year={term.termName} classList={classes} id={term.termName}></Term>
+                  <Term year={term.termName} classList={term.classes.map((flowchartClass: FlowchartClass) => classDB[flowchartClass.id])} totalUnits={term.totalUnits || 0} id={term.termName} />
+                  {index === flowchart.length - 1 && (
```
Comment (Mitashi-11) [inline src/frontend/src/Components/Grid.tsx:114]: This adds the total units at the end of the last term. Could it be moved to another layer on top of the last term or some other spot on the screen?

File: src/frontend/src/Components/Grid.tsx [hunk 2]
```diff
@@ -60,16 +70,19 @@ function Grid() {
       try {
         const termClassData: { [ClassId: string]: ClassDBClass } = {};
         const promises = flowchart.map(async (term) => {
+          let termTotalUnits = 0;
           await Promise.all(
             term.classes.map(async (flowchartClass: FlowchartClass) => {
               const response: AxiosResponse<QuarterClassData> =
                 await axios.get('http://localhost:8080/get/QuarterClass/' + flowchartClass.id);
```
Comment (Mitashi-11) [inline src/frontend/src/Components/Grid.tsx:77]: Is there a more efficient way to do this? The await inside the promise.all might be ineffective when there are many promises.

**Review by Mitashi-11 — CHANGES_REQUESTED** [review]

File: src/frontend/src/Components/Grid.tsx (hunk 1 above)
Comment (jluna28) [reply src/frontend/src/Components/Grid.tsx]: done

File: src/frontend/src/Components/Grid.tsx (hunk 2 above)
Comment (jluna28) [reply src/frontend/src/Components/Grid.tsx:77]: await Promise.all was not part of this PR, will make another issue to address

**Review by Mitashi-11 — APPROVED** [review]

**Pull request merged**